compiler/kernel/expr.go

```diff
@@ -350,13 +350,14 @@ func compileCall(zctx *zed.Context, call dag.Call) (expr.Evaluator, error) {
 		return compileShaper(zctx, call)
 	}
 	nargs := len(call.Args)
-	fn, this, err := function.New(zctx, call.Name, nargs)
+	fn, wantPath, err := function.New(zctx, call.Name, nargs)
```

Comment:
Nit: `wantPath` reads like a Boolean. I'd call this `path` (and call the `dag.Path` below `dagPath`).